internal/core/data/v2/application/subscriber.go

```diff
@@ -60,6 +62,11 @@ func SubscribeEvents(ctx context.Context, dic *di.Container) errors.EdgeX {
 					lc.Errorf("fail to unmarshal event, %v", err)
 					break
 				}
+				err = validateEvent(msgEnvelope.ReceivedTopic, event.Event)
```

Comment:
Nice use of msgEnvelope.ReceivedTopic that I had not thought of when I added it.